### ISSUE TITLE
Undesired RIA URL reconfiguration

### ISSUE THREAD

**Issue by mih** [datalad/datalad#5628]
I have a superdataset with

```
% datalad x-configuration | grep subdataset | head -1
datalad.get.subdataset-source-candidate-100origin=ria+ssh://bulk1.htc.inm7.de/ds/studyforrest/srv#{id}
```

because I want all subdataset to be cloned from an internal store, despite `.gitmodules` pointing to public sources.

That in turn causes `clone` to reconfigure the access URL for an ORA special remote to use this store URL. However, the ORA remote is already configured properly have this as `pushurl`, but uses a more widely accessible, different `http` URL from read-access. This configuration is replaced by a uniform SSH setup. Whenever I push back from this clone to the store, all `http` access to this remote is lost.

ping @bpoldrack

**Comment by bpoldrack** [comment]
FTR: Two things instantly coming to mind:
1. Reconfiguration needs double-check to account for the "new" url/push-url distinction
2. Still strange, since reconfiguration is supposed to only happen, if the stored access method fails.

**Comment by mih** [comment]
Sounds sensible. From the top of my head I do not know when we would ever want to commit.

**Comment by bpoldrack** [comment]
Cool. Since local special remote configs are now officially supported by annex, will switch entirely.

**Comment by mih** [comment]
It will be critical to check which annex version is required for that.

Also, a simple and _quick_ fix first, and an improvement later would be much appreciated. I already spend hours undoing the goodness

**Comment by bpoldrack** [comment]
Not sure there is a quick fix, though. We explicitly introduced the idea, that we do want a reconfiguration, if the URL of the ORA remote does not match the clone URL, despite both belonging to the same store. I could introduce a special case where we say: "If one of `url`/`push-url` matches, don't touch it", but while this may solve this particular case, it's still undesireable in another. Say you clone that dataset locally, then you would still go through HTTP to retrieve data in that clone (while ATM you'd get a reconfiguration giving you local access).

**Comment by bpoldrack** [comment]
There's another aspect. This doesn't seem true to me:

> Whenever I push back from this clone to the store, all http access to this remote is lost.

If anyone clones the changed dataset from that store via HTTP, they should get a reconfiguration to HTTP, too. While that committed ping-pong remains annoying, there's no data access loss, that I can see.

**Comment by bpoldrack** [comment]
> I could introduce a special case where we say: "If one of url/push-url matches, don't touch it"

FTR: Doing this as the desired quick fix. Currently we check in `clone`, whether there are only enabled ORA remotes, that don't match the clone URL. If that's the case, we check whether anyone of them actually does point to the same store and if so, reconfigure it to use the clone URL for ORA, too. So, if there's a match - no reconfigure. However, we only check against `url`, not considering `push-url`. With the "fix" we consider both `url` and `push-url` for matching the clone URL, leading to no reconfiguration if the clone URL is different from the special remote's `url`, but not different from `push-url`. However, as mentioned above, that solves the immediate issue, but not the reconfig problem in general, and this behavior is what we wouldn't want to happen in other cases.

Ultimate solution has to be local config, I think.

**Comment by mih** [comment]
Sounds good. Will be clearer in code, I think.

Thx!

**Comment by mih** [comment]
Can this be closed after #5672 ?
